src/main/java/com/ldtteam/structurize/Structurize.java

```diff
@@ -42,9 +44,23 @@ public Structurize()
         Mod.EventBusSubscriber.Bus.FORGE.bus().get().register(EventSubscriber.class);
         DistExecutor.unsafeRunWhenOn(Dist.CLIENT, () -> () -> Mod.EventBusSubscriber.Bus.FORGE.bus().get().register(ClientEventSubscriber.class));
 
-        if (DataVersion.CURRENT == DataVersion.UPCOMING)
+        if (DataFixesManager.getDataFixer() instanceof com.mojang.datafixers.DataFixerUpper)
         {
-            throw new RuntimeException("Missing some newest data versions. Please update com/ldtteam/structures/blueprints/v1/DataVersion");
+            if (DataFixesManager.getDataFixer().getSchema(Integer.MAX_VALUE - 10).getVersionKey() >= DataVersion.UPCOMING.getDataVersion())
+            {
+                throw new RuntimeException("Missing some newest data versions. Please update com/ldtteam/structures/blueprints/v1/DataVersion");
```

Comment:
or something like that, the first sentence can be as long as `Missing some newest data versions. Please update` and should tell players to check their versions
```suggestion
                throw new RuntimeException("You are trying to run old mod on much newer vanilla. Missing some newest data versions. Please update com/ldtteam/structures/blueprints/v1/DataVersion");
```